File: C++ for Everyone, 2nd Ed, Horstmann/Chapter4/P4-3C.cpp
//Write programs that read a line of input as a string and print
//c.The string, with all vowels replaced by an underscore.

#include<iostream>
#include<vector>

using namespace std;

int main()
{
    cout << "Enter a string: ";
    string input;
    getline(cin, input);

    vector<char> inputCstyle (input.length() + 1);
    copy(input.begin(), input.end(), inputCstyle.begin());

    for(int i = 0; i < input.length() + 1; i++)
    {
        if(inputCstyle[i] == 'A' || inputCstyle[i] == 'a' || inputCstyle[i] == 'E' || inputCstyle[i] == 'e' || inputCstyle[i] == 'O' || inputCstyle[i] == 'o' || inputCstyle[i] == 'U' || inputCstyle[i] == 'u' || inputCstyle[i] == 'I' || inputCstyle[i] == 'i')
        {
            inputCstyle[i] = '_';
        }
    }

    for(int j = 0; j < input.length() + 1; j++)
    {
        cout << inputCstyle[j];
    }

    return 0;
}

File: Python-For-Everyone-Horstmann/Chapter4-Loops/P4.3C.py
# Write programs that read a line of input as a string and print
#     a.  Only the uppercase letters in the string.
#     b.  Every second letter of the string.
#     c.  The string, with all vowels replaced by an underscore.
#     d.  The number of digits in the string.
#     e.  The positions of all vowels in the string.

inputString = str(input("Enter a string: "))
outputString = ""
# saving time writing long statements
vowels = ('a','e','i','o','u','A','E','I','O','U')

for i in range(len(inputString)):
    if inputString[i] in vowels:
        outputString += "_"

    else:
        outputString += inputString[i]

print(outputString)